Todor efcore ds (#2168)
telerik/reporting-docs: designing-reports/connecting-to-data/data-source-components/entitycoredatasource-component/connecting-to-a-dbcontext-with-the-entitycoredatasource-component.md

The wizard walks you through the process as explained in:
* [Standalone Report Designer for .NET EntityCoreDataSource Wizard](slug:desktop-entitycoredatasource-wizard)
* [Web Report Designer EntityCoreDataSource Wizard](slug:wrd-entitycoredatasource-wizard)

After the wizard completes, the component appears in the [Data Explorer](slug:telerikreporting/designing-reports/report-designer-tools/desktop-designers/tools/data-explorer) with the schema of the selected entity, and the report data items can bind to its fields.

## Configuring the Component Programmatically

The `EntityCoreDataSource` class exposes three constructors:

```CSharp
public EntityCoreDataSource();
public EntityCoreDataSource(object context, string contextMember);
public EntityCoreDataSource(string connectionString, object context, string contextMember);
```

> The simplest configuration uses the parameterless constructor and assigns at least the `Context` and `ContextMember` properties. Assign the `DbContext` type to `Context` and the name of the `DbSet<T>`, queryable property, or method to `ContextMember`:

Reviewer: ```suggestion
To create the simplest configuration, use the parameterless constructor and set the `Context` and `ContextMember` properties. Set `Context` to the `DbContext` type and `ContextMember` to the name of the `DbSet<T>`, queryable property, or method that retrieves the data:
```

```CSharp
var dataSource = new Telerik.Reporting.EntityCoreDataSource
{
    Context = typeof(AppDbContext),
    ContextMember = "People"
};
```

> Use the three-argument constructor when you also need to set a connection string in one expression:

Reviewer: ```suggestion
Use the three-argument constructor when you need to set the connection string, context type, and context member in a single call:
```

```CSharp
var dataSource = new Telerik.Reporting.EntityCoreDataSource(
	"Server=.;Database=MSSQLLocalDB;Integrated Security=True;TrustServerCertificate=True",
	typeof(AppDbContext),
	"People");
```

## Binding to a DbContext Type Versus an Instance

The `Context` property is typed as `object` and accepts either a `Type` reference or a live `DbContext` instance:

- When you supply a **type**, the component instantiates the `DbContext`, holds it for the duration of report processing, and disposes it automatically. This is the recommended pattern because it preserves [lazy loading](slug:entitycoredatasource-context-lifecycle) for the duration of report processing.
- When you supply an **instance**, the application owns the lifetime of the context. The component does not call `Dispose` on the supplied instance.

```CSharp
var context = new AppDbContext(connectionString);

var dataSource = new Telerik.Reporting.EntityCoreDataSource
{
    Context = context,
    ContextMember = "People"
};

> // You have to dispose the context explicitly when done with the report.

Reviewer: ```suggestion
// You have to dispose the context manually after rendering the report.
```
